### DISCUSSION TITLE
How to add extensions

### DISCUSSION THREAD

**Discussion by penielny** [ultrafunkamsterdam/undetected-chromedriver#2045] [Q&A]
Hi, guys I'm finding it difficult to add an extension to my project any help ??
the code i have below is not working and sugestions
`
 def set_up_driver(self):
        chrome_options = uc.ChromeOptions()
        random_user_agent = random.choice(self.user_agents)
        chrome_options.add_argument(f"--user-agent={random_user_agent}")
        chrome_options.add_argument("--disable-dev-shm-usage")
        chrome_options.add_argument("--disable-gpu")
        chrome_options.add_argument("--no-sandbox")
        chrome_options.add_argument("--disable-blink-features=AutomationControlled")

        # Add proxy extension if needed
        if self.proxy:
            extension = self.getProxyExtenstion()
            try:
                chrome_options.add_extension(extension)
            except Exception as e:
                print(f"Failed to load extension: {e}")

        # Initialize undetected-chromedriver
        try:
            self.driver = uc.Chrome(headless=False, use_subprocess=False, options=chrome_options)
            self.wait = WebDriverWait(self.driver, 10)
        except Exception as e:
            print(f"Error initializing driver: {e}")

`

**Comment by ddavis-ssc** [comment]
I went through this process not too long ago.

Personally, I could never get chrome_options.add_extension() to work, I use:

`chrome_options.add_argument("--load-extension=Extensions\\proxy_auth_plugin_"+nprx)`

I have boiler plate templates for manifest.js, content.js, etc. and when I call a driver to open with a proxy it checks if the appropriate extension has been created, if it hasn't been created then I create it leveraging my proxy details and my boiler plate js scripts.

**Comment by penielny** [comment]
Really, Thanks man this is very helpful.
